docs/unit-testing-for-monitoring-metrics.md

```diff
@@ -0,0 +1,71 @@
+
+# Testing ARO Monitor Metrics
+
+
+
+## The Monitor Architecture
+
+The ARO monitor component (the part of the aro binary you activate when you execute ./cmd/aro monitor) collects and emits the various metrics about cluster health (and its own) we want to see in Geneva. 
+
+To send data to Geneva the monitor uses an instance of a Geneva MDM container as a proxy of the Geneva API. The MDM container accepts statsd formatted data (the Azure Geneva version of statsd, that is) over a UNIX (Domain) socket. The MDM container then forwards the metric data over a https link to the Geneva API. Please note that using a Unix socket can only be accessed from then same machine. 
+
+The monitor picks the required information about which clusters should actualyl monitor from its corresponding Cosmos DB. If multiple monitor instances run in parallel  (i.e. connect to the same database instance) as is the case in production, they negotiate which instance monitors what cluster (see : [monitoring.md](./monitoring.md)). 
+
+
+![Aro Monitor Architecture](img/AROMonitor.png "Aro Monitor Architecture")
```

Comment:
I'd move the image to line 9 between the first and second paragraph.
From the text it's not obvious that Geneva MDM container is a different component than Geneva that serves the API. Official naming doesn't help here :/